seed/views/v3/organizations.py

```diff
@@ -5,29 +5,239 @@
+
+        # We don't wish to double count sub organization memberships.
+        org_users = OrganizationUser.objects.select_related('user').only(
+            'role_level', 'user__first_name', 'user__last_name', 'user__email', 'user__id'
+        ).filter(organization=o)
+
+        owners = []
+        role_level = None
+        user_is_owner = False
+        for ou in org_users:
+            if ou.role_level == ROLE_OWNER:
+                owners.append({
+                    'first_name': ou.user.first_name,
+                    'last_name': ou.user.last_name,
+                    'email': ou.user.email,
+                    'id': ou.user.id
+                })
+
+                if ou.user == request.user:
+                    user_is_owner = True
+
+            if ou.user == request.user:
+                role_level = _get_js_role(ou.role_level)
+
+        org = {
+            'name': o.name,
+            'org_id': o.id,
+            'id': o.id,
+            'number_of_users': len(org_users),
+            'user_is_owner': user_is_owner,
+            'user_role': role_level,
+            'owners': owners,
+            'sub_orgs': _dict_org(request, o.child_orgs.all()),
+            'is_parent': o.is_parent,
+            'parent_id': o.parent_id,
+            'display_units_eui': o.display_units_eui,
+            'display_units_area': o.display_units_area,
+            'display_significant_figures': o.display_significant_figures,
+            'cycles': cycles,
+            'created': o.created.strftime('%Y-%m-%d') if o.created else '',
+            'mapquest_api_key': o.mapquest_api_key or '',
+            'display_meter_units': o.display_meter_units,
+            'thermal_conversion_assumption': o.thermal_conversion_assumption,
+        }
+        orgs.append(org)
+
+    return orgs
+
+
+def _dict_org_brief(request, organizations):
+    """returns a brief dictionary of an organization's data."""
+
+    organization_roles = list(OrganizationUser.objects.filter(user=request.user).values(
+        'organization_id', 'role_level'
+    ))
+
+    role_levels = {}
+    for r in organization_roles:
+        role_levels[r['organization_id']] = _get_js_role(r['role_level'])
+
+    orgs = []
+    for o in organizations:
+        user_role = None
+        try:
+            user_role = role_levels[o.id]
+        except KeyError:
+            pass
+
+        org = {
+            'name': o.name,
+            'org_id': o.id,
+            'parent_id': o.parent_org_id,
+            'id': o.id,
+            'user_role': user_role
+        }
+        orgs.append(org)
+
+    return orgs
+
+
+def _get_js_role(role):
+    """return the JS friendly role name for user
+
+    :param role: role as defined in superperms.models
+    :returns: (string) JS role name
+    """
+    roles = {
+        ROLE_OWNER: 'owner',
+        ROLE_VIEWER: 'viewer',
+        ROLE_MEMBER: 'member',
+    }
+    return roles.get(role, 'viewer')
+
+
+def _get_role_from_js(role):
```

Comment:
I think this isn't used. Can you confirm and delete if so?

Reply:
Good find 👍